src/nix/flake.cc

```diff
@@ -334,10 +334,10 @@ struct CmdFlakeCheck : FlakeCommand
             try {
                 state->forceValue(v, pos);
                 if (!v.isLambda() || v.lambda.fun->matchAttrs || std::string(v.lambda.fun->arg) != "final")
-                    throw Error("overlay does not take an argument named 'final'");
+                    throw Error("By convention the attribute 'overlay' expects the argument 'final' not '" + std::string(v.lambda.fun->arg) + "'");
```

Comment:
This will crash if `v` is not a lambda.

Reply:
In the new commit the `!v.isLambda()` check is now isolated, and also throws a more appropriate error message, telling the user the attribute should be a lambda.